.github/workflows/build-container.yml

```diff
@@ -11,30 +11,61 @@ jobs:
-      DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-      DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
       INSP_VERSION: v3.10.0
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Prepare environment
+      - 
+        uses: actions/checkout@v2
+      - 
+        name: Prepare environment
         run: |
           sudo apt-get -qq update
           sudo apt-get -y -o Dpkg::Options::="--force-confnew" install moby-engine moby-cli
           sudo apt-get install -y openssl parallel
-
-      - name: Build container
-        run: |
-          docker swarm init --advertise-addr `ip addr s | grep global | grep -oE '((1?[0-9][0-9]?|2[0-4][0-9]|25[0-5])\.){3}(1?[0-9][0-9]?|2[0-4][0-9]|25[0-5])' | head -n1`
-          docker build --build-arg VERSION="${INSP_VERSION}" -t inspircd:testing .
-
-      - name: Push container
+      -
+        name: Docker meta
         if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
-        run: |
-          if [ "$DOCKER_USERNAME" != "" ] && [ "$DOCKER_PASSWORD" != "" ]; then
-            ./tools/tagging.sh "${DOCKER_REPO_PREFIX}/${DOCKER_REPO_NAME}" "${INSP_VERSION:-latest}" ""
-            docker images "${DOCKER_REPO_PREFIX}/${DOCKER_REPO_NAME}"
-            echo "$DOCKER_PASSWORD" | docker login --username "$DOCKER_USERNAME" --password-stdin
-            docker images --format "{{.Repository}}:{{.Tag}}" "${DOCKER_REPO_PREFIX}/${DOCKER_REPO_NAME}" | xargs -L 1 docker push
-            docker run --rm -v "$(pwd)/:/data/:ro" -e "DOCKERHUB_USERNAME=$DOCKER_USERNAME" -e "DOCKERHUB_PASSWORD=$DOCKER_PASSWORD" -e "DOCKERHUB_REPO_PREFIX=$DOCKER_REPO_PREFIX" -e "DOCKERHUB_REPO_NAME=$DOCKER_REPO_NAME" sheogorath/readme-to-dockerhub
-          fi
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.DOCKER_REPO_PREFIX }}/${{ env.DOCKER_REPO_NAME }}
+          tags: |
+            type=semver,pattern={{version}},value=${{ env.INSP_VERSION }}
+            type=semver,pattern={{major}}.{{minor}},value=${{ env.INSP_VERSION }}
+            type=semver,pattern={{major}},value=${{ env.INSP_VERSION }}
+      - 
+        name: Set up QEMU
+        if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
+        uses: docker/setup-qemu-action@v1
+      -
+        name: Set up Docker Buildx
+        if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Login to DockerHub
+        if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      -
+        name: Build and push
+        if: github.repository == 'inspircd/inspircd-docker' && github.ref == 'refs/heads/master'
+        id: docker_build
```

Comment:
Is this id used anywhere?

Reply:
no don't think so - it's pretty much copied and pasted from the action docs. will remove.